spring-data-relational/src/main/java/org/springframework/data/relational/core/sql/UpdateBuilder.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.relational.core.sql;
+
+/**
+ * Entry point to construct an {@link Update} statement.
+ *
+ * @author Mark Paluch
+ * @since 1.1
+ * @see StatementBuilder
+ */
+public interface UpdateBuilder {
+
+	/**
+	 * Configure the {@link Table} to which the update is applied.
+	 *
+	 * @param count the top count.
+	 * @return {@code this} {@link SelectBuilder}.
+	 */
+	UpdateAssign table(Table table);
+
+	/**
+	 * Interface exposing {@code SET} methods.
+	 */
+	interface UpdateAssign {
+
+		/**
+		 * Apply a {@link Assignment SET assignment}.
+		 *
+		 * @param assignment a single {@link Assignment column assignment}.
+		 * @return {@code this} builder.
+		 * @see Assignment
+		 */
+		UpdateAssignAnd set(Assignment assignment);
+	}
+
+	/**
+	 * Interface exposing {@code SET} methods.
+	 */
+	interface UpdateAssignAnd extends UpdateWhere {
+
+		/**
+		 * Apply a {@link Assignment SET assignment}.
+		 *
+		 * @param assignment a single {@link Assignment column assignment}.
+		 * @return {@code this} builder.
+		 * @see Assignment
+		 */
+		UpdateAssignAnd and(Assignment assignment);
```

Comment:
I find a series of `set` calls more intuitive than `and`.
How about offering `set` as an alias to `and`?

We even could go for a vararg:

```
Update update = StatementBuilder.update(table)
    .set(
        foo.set(SQL.bindMarker()),
        bar.set(SQL.bindMarker())
    ).build();

```

Reply:
`and(…)` and `or(…)` syntax aligns with the where and join syntax. We can accept varargs here, probably, also a `Collection<? extends Assignment>`.

Reply:
The difference to joins and where syntax I see is that in SQL there is no `AND` between assignments.

Reply:
Makes sense.